packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3419,17 +3419,20 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   /// until it finds a [RenderObjectElement].
   RenderObject? get renderObject {
     RenderObject? result;
-    void visit(Element element) {
-      assert(result == null); // this verifies that there's only one child
-      if (element._lifecycleState == _ElementLifecycle.defunct) {
-        return;
-      } else if (element is RenderObjectElement) {
-        result = element.renderObject;
+    final Queue<Element> elements = Queue<Element>();
```

Comment:
I've generally found that data structures besides list/map/set are not particularly fast in Dart.  I'm curious if you tried with just a plain List

Reply:
I couldn't use a List because "removeFirst" would be O(n).  In order to use a List directly I'd have to do something like a RingBuffer, but then there is the risk of overflow.

Reply:
Fair enough, I just checked and the default Queue in dart is a ListQueue with a resizing ring buffer. No need to re-implement.